lib/ttnt/test_selector.rb

```diff
@@ -29,6 +32,8 @@ def select_tests
       diff = @base_obj.diff(@target_obj)
       diff.each_patch do |patch|
         file = patch.delta.old_file[:path]
+        tests << file and next if test_file?(file)
```

Comment:
What do you think about making an explicit `if` / `else` like:

~~~ruby
if test_file?(file)
  tests << file
else
  patch.each_hunk do |hunk|
    hunk.each_line do |line|
      case line.line_origin
      when :addition
        tests += mapping.get_tests(file: file, lineno: line.new_lineno)
      when :deletion
        tests += mapping.get_tests(file: file, lineno: line.old_lineno)
      end
    end
  end
end
~~~

We easily see that we have two branches, one for each kind of files. Also feel free to ask me regarding your `FIXME` and `TODO` comments ; I do not understand them. :p